test/test_reductions.py

```diff
@@ -814,6 +814,13 @@ def test_prod(self, device, dtype):
         torch.prod(x, 1, out=res2)
         self.assertEqual(res1, res2)
 
+    def test_prod_bool(self, device):
+        vals = [[True, True], [True, False], [False, False], []]
+        for val in vals:
```

Comment:
Would you extend this test to verify bool->int64 consistent with NumPy, too?

Reply:
Addressed! Thanks for suggestion!